designsystem/src/commonMain/kotlin/org/hisp/dhis/mobileui/designsystem/component/RadioButton.kt

```diff
@@ -40,51 +40,37 @@ fun RadioButton(
     onClick: () -> Unit
 ) {
     CompositionLocalProvider(LocalRippleTheme provides CustomDHISRippleTheme) {
-        if (textInput.isNullOrEmpty()) {
+
+        Row(
+            modifier = Modifier.width(Spacing.Spacing840).height(Spacing.Spacing40),
+            horizontalArrangement = Arrangement.spacedBy(0.dp, Alignment.Start),
+            verticalAlignment = Alignment.Top
+        ) {
             RadioButton(
                 selected = selected,
                 onClick = onClick,
                 enabled = enabled,
                 modifier = Modifier
-                    .size(Spacing.Spacing40)
-                    .padding(Spacing.Spacing8),
+                    .size(Spacing.Spacing40),
                 colors = RadioButtonDefaults.colors(
                     selectedColor = SurfaceColor.Primary,
                     unselectedColor = Outline.Dark,
                     disabledSelectedColor = TextColor.OnDisabledSurface,
                     disabledUnselectedColor = TextColor.OnDisabledSurface
                 )
             )
-        } else {
-            Row(
-                modifier = Modifier.width(840.dp).height(Spacing.Spacing40),
-                horizontalArrangement = Arrangement.spacedBy(0.dp, Alignment.Start),
-                verticalAlignment = Alignment.Top
-            ) {
-                RadioButton(
-                    selected = selected,
-                    onClick = onClick,
-                    enabled = enabled,
-                    modifier = Modifier
-                        .size(Spacing.Spacing40),
-                    colors = RadioButtonDefaults.colors(
-                        selectedColor = SurfaceColor.Primary,
-                        unselectedColor = Outline.Dark,
-                        disabledSelectedColor = TextColor.OnDisabledSurface,
-                        disabledUnselectedColor = TextColor.OnDisabledSurface
-                    )
-                )
+            if (!textInput.isNullOrEmpty()) {
                 Row(
```

Comment:
Is this row with only one item needed, could be removed and set only Text component?

Reply:
I have followed the Sigma structure, but it can be changed